husky_components/dropdown/main.js

```diff
@@ -151,36 +154,61 @@ define([], function() {
         },
 
         itemEnable: function(id) {
-            this.sandbox.dom.removeClass(this.$find('li[data-id="'+ id +'"]'), 'disabled');
+            this.sandbox.dom.removeClass(this.$find('li[data-id="' + id + '"]'), 'disabled');
         },
 
         itemDisable: function(id) {
-            this.sandbox.dom.addClass(this.$find('li[data-id="'+ id +'"]'), 'disabled');
+            this.sandbox.dom.addClass(this.$find('li[data-id="' + id + '"]'), 'disabled');
         },
 
         // trigger event with clicked item
         clickItem: function(id) {
-            this.sandbox.util.foreach(this.options.data, function(item) {
-                if (typeof item.id !== 'undefined' && item.id.toString() === id.toString()) {
-                    this.sandbox.logger.log(this.name, 'item.click: ' + id, 'success');
+            var item = this.getItemById(id),
+                $item = this.$el.find('*[data-id="' + id + '"]');
 
-                    if (!!item.callback && typeof item.callback === 'function') {
-                        item.callback.call(this);
-                    } else if (!!this.options.clickCallback && typeof this.options.clickCallback === 'function') {
-                        this.options.clickCallback(item, this.$el);
-                    } else {
-                        this.sandbox.emit(this.getEvent('item.click'), item, this.$el);
-                    }
+            if (!item) {
+                return;
+            }
+
+            if (!!item.callback && typeof item.callback === 'function') {
+                item.callback.call(this);
+            } else if (!!this.options.clickCallback && typeof this.options.clickCallback === 'function') {
+                this.options.clickCallback(item, this.$el);
+            } else {
+                this.sandbox.emit(this.getEvent('item.click'), item, this.$el);
+            }
+
+            if (!item[this.options.clickedName]) {
+                return this.hideDropDown();
+            }
+
+            $item.addClass('highlight-animation');
+            $item.find('.info, .info-clicked').addClass('clicked');
+
+            _.delay(function() {
+                $item.removeClass('highlight-animation');
+                $item.find('.info, .info-clicked').removeClass('clicked');
```

Comment:
I would save the result from the call a few lines earlier instead of calling `find` again.